File: core/src/main/java/com/victorlh/android/framework/core/dialog/AbstractProgressDialog.java
package com.victorlh.android.framework.core.dialog;

import android.content.Context;
import android.widget.ProgressBar;

import androidx.annotation.Nullable;

/**
 * @author Victor
 * 09/06/2018
 */
public abstract class AbstractProgressDialog extends AbstractDialog implements IProgressDialog {

	protected int progresoMaximo;

	public AbstractProgressDialog(Context context) {
		super(context);
	}

	@Nullable
	protected abstract ProgressBar getProgressBar();

	@Override
	protected void initValues() {
		super.initValues();
		setCancelable(false);
	}

	@Override
	public void setProgreso(int progreso) {
		ProgressBar progressBar = getProgressBar();
		if (progressBar != null) {
			progressBar.setProgress(progreso);
		}
	}

	@Override
	public void setProgresoMaximo(int progresoMaximo) {
		this.progresoMaximo = progresoMaximo;
	}

}

File: core/src/main/java/com/victorlh/android/framework/core/dialog/AbstractProgressDialog.kt
package com.victorlh.android.framework.core.dialog

import android.content.Context
import android.widget.ProgressBar

/**
 * @author Victor
 * 09/06/2018
 */
abstract class AbstractProgressDialog(context: Context) : AbstractDialog(context), IProgressDialog {

	init {
		cancelable = false
	}

	override var progresoMaximo = 0
	protected abstract val progressBar: ProgressBar?
	override var progreso: Int = 0
		get() = field
		set(value) {
			progressBar?.progress = value
		}


}